### PR TITLE
Fix bugs in MNIST implementation

### LINKED ISSUE #90: Model inference
Check on the MNIST model inference - prediction for test digits.

Test Procedure
1. Train MNIST model locally on 10 digits (as attached below): 0-9 with result 100% training and  100% validation accuracy 
2. Save model by clicking on button
3. Test model with digits used during training

Resulting CSV File output for all digits (uploaded in order smallest to largest digit) is attached to this post.
Screenshot 2021-06-21 at 15 40 33

Digits used for training and testing:
Screenshot 2021-06-21 at 14 48 39

9-filled-32

Set up: tested on local machine, nvm v16.3.0, npm 7.15.1

Console log 
[Log] Start: Processing Uploaded File 
[Log] User File Validated. Start parsing.
[Log] Start Training
[Log] _________________________________________________________________
[Log] Layer (type)                 Output shape              Param #   
[Log] =================================================================
[Log] conv2d_Conv2D1 (Conv2D)      [null,26,26,16]           448       
[Log] _________________________________________________________________
[Log] max_pooling2d_MaxPooling2D1  [null,13,13,16]           0         
[Log] _________________________________________________________________
[Log] conv2d_Conv2D2 (Conv2D)      [null,11,11,32]           4640      
[Log] _________________________________________________________________
[Log] max_pooling2d_MaxPooling2D2  [null,5,5,32]             0         
[Log] _________________________________________________________________
[Log] conv2d_Conv2D3 (Conv2D)      [null,3,3,32]             9248      
[Log] _________________________________________________________________
[Log] flatten_Flatten1 (Flatten)   [null,288]                0         
[Log] _________________________________________________________________
[Log] dense_Dense1 (Dense)         [null,64]                 18496     
[Log] _________________________________________________________________
[Log] dense_Dense2 (Dense)         [null,10]                 650       
[Log] =================================================================
[Log] Total params: 33482
[Log] Trainable params: 33482
[Log] Non-trainable params: 0
[Log] _________________________________________________________________
[Log] Proxy
[Log] EPOCH (1): Train Accuracy: 100.00,
             Val Accuracy:  100.00

[Log] loss 0.1054
[Log] Proxy
[Log] EPOCH (2): Train Accuracy: 100.00,
             Val Accuracy:  100.00

[Log] loss 0.1054
[Log] Proxy
[Log] EPOCH (3): Train Accuracy: 100.00,
             Val Accuracy:  100.00

[Log] loss 0.1054
[Log] Proxy
[Log] EPOCH (4): Train Accuracy: 100.00,
             Val Accuracy:  100.00

[Log] loss 0.1054
[Log] Proxy
[Log] EPOCH (5): Train Accuracy: 100.00,
             Val Accuracy:  100.00

[Log] loss 0.1054
[Log] Proxy
[Log] EPOCH (6): Train Accuracy: 100.00,
             Val Accuracy:  100.00

[Log] loss 0.1054
[Log] Proxy
[Log] EPOCH (7): Train Accuracy: 100.00,
             Val Accuracy:  100.00

[Log] loss 0.1054
[Log] Proxy
[Log] EPOCH (8): Train Accuracy: 100.00,
             Val Accuracy:  100.00

[Log] loss 0.1054
[Log] Proxy
[Log] EPOCH (9): Train Accuracy: 100.00,
             Val Accuracy:  100.00

[Log] loss 0.1054
[Log] Proxy
[Log] EPOCH (10): Train Accuracy: 100.00,
             Val Accuracy:  100.00

[Log] loss 0.1054
[Log] mnist-model
[Log] Deactivated
[Log] <input id="hidden-input" type="file" multiple class="hidden">

[Log] Loading model...
[Log] Model loaded.
[Log] Prediction Sucessful!
[Log] undefined
[Log] Object`

### PR DESCRIPTION
**Pull request by mmilenkoski**
This PR resolves #90 

With these changes, the model should no longer reach 100% percent validation and train accuracy. It should also be able to produce decent results on images that were used for training when you upload them in the test phase (if they were not selected for the validation set). 

However, it tends to overfit the training set and the validation accuracy is 0. We should look into this next. 

We had several issues:
1. We were using different names to save the trained model and to load a model in the test component so the test results were very different than the training ones. 
2. We were using wrong loss - binary instead of categorical crossentropy
3. We were not storing the labels correctly

I also optimized the preprocessing to dispose of intermediate tensors.

### CODE REVIEW

**Review by martinjaggi — APPROVED** [review]
nice!
